### PR TITLE
Workflow for publishing npm package

### PR DESCRIPTION
**Pull request by hirenf14**

### CODE REVIEW

**Comment by hirenf14** [comment]
Hi @alicanerdurmaz , 
I have tried to add workflow using `changesets` to manage releases to npm registry.
Can you help me review this if its ok or not?

Thanks.

File: .changeset/config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.3.1/schema.json",
+  "changelog": ["@changesets/changelog-github", { "repo": "refinedev/refine" }],
```
Comment (alicanerdurmaz) [inline .changeset/config.json:3]: is this `"repo": "refinedev/refine"` a typo? 

```suggestion
  "changelog": ["@changesets/changelog-github", { "repo": "refinedev/refine" }],
```

Comment (hirenf14) [reply .changeset/config.json:3]: yes, its typo. Thanks for catching it.

Comment (hirenf14) [reply .changeset/config.json]: fixed.

**Comment by omeraplak** [comment]
LGFM! 💥

**Comment by hirenf14** [comment]
> LGFM! 💥

can you approve it? I have restriction over the repo, to merge I need 1 reviewer to approve.

**Review by omeraplak — APPROVED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

**Pull request merged**